core/src/peer_id.rs

```diff
@@ -35,7 +35,7 @@ const MAX_INLINE_KEY_LENGTH: usize = 42;
 // TODO: maybe keep things in decoded version?
 #[derive(Clone, Eq)]
 pub struct PeerId {
-    multihash: Multihash,
+    multihash: Vec<u8>,
```

Comment:
```suggestion
    multihash: Bytes,
```
It would be good to retain the cheap cloning of `PeerId`s, as these are cloned a lot. See also https://github.com/libp2p/rust-libp2p/pull/1187.

Reply:
Done.